pixeltable/catalog/table_version.py

```diff
@@ -545,14 +554,25 @@ def _add_index(self, col: Column, idx_name: Optional[str], idx: index.IndexBase)
         idx_info = self.IndexInfo(id=idx_id, name=idx_name, idx=idx, col=col, val_col=val_col, undo_col=undo_col)
         self.idx_md[idx_id] = idx_md
         self.idxs_by_name[idx_name] = idx_info
+        try:
+            idx.create_index(self._store_idx_name(idx_id), val_col)
+        finally:
+            # Remove in memory index structure in the case of error
+            def remove_index() -> None:
+                del self.idxs_by_name[idx_name]
+                del self.idx_md[idx_id]
+                self.next_idx_id = idx_id
 
+            run_cleanup_on_exception(remove_index)
+
+    def _add_index(self, col: Column, idx_name: Optional[str], idx: index.IndexBase) -> UpdateStatus:
```

Comment:
refactored this into 3 parts 
create index columns
add columns
create index along with its md